internal/controller/nim/account_controller.go

```diff
@@ -383,6 +448,41 @@ func (r *AccountReconciler) reconcileNimPullSecret(
 	return secret, nil
 }
 
+// getSelectedModelList returns a list of Ids of the selected models
+func (r *AccountReconciler) getSelectedModelList(
+	ctx context.Context, cmRef *corev1.ObjectReference,
+	namespace string) ([]string, error) {
+	// if selected model list is not set
+	if cmRef == nil || len(cmRef.Name) == 0 {
+		return []string{}, nil
+	}
+
+	// get the config map that contains the selected model list
+	cmNs := cmRef.Namespace
+	if cmNs == "" {
+		cmNs = namespace
+	}
+	modelListCm := &corev1.ConfigMap{}
+	modelListCmSubject := types.NamespacedName{Name: cmRef.Name, Namespace: cmNs}
+	if err := r.Client.Get(ctx, modelListCmSubject, modelListCm); err != nil {
+		return nil, err
+	}
+
+	// convert the selected model list to a string array
+	if modelListCm.Data != nil {
+		if models, ok := modelListCm.Data["models"]; ok {
+			var selectedModelIds []string
+			err := json.Unmarshal([]byte(models), &selectedModelIds)
+			if err != nil {
+				return nil, err
+			}
+			return selectedModelIds, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to get the selected model list from the config map %+v", modelListCmSubject)
```

Comment:
Much like the comment above, if the user had a wrong syntax in the ConfigMap, returning an error will requeueu and we'll be right back here. It will be better to log the error and not return it, so the rest of the flow will work as if this ConfigMap didn't exist.